core/src/main/java/com/linecorp/armeria/common/logging/TextLogFormatter.java

```diff
@@ -0,0 +1,254 @@
+            responseCauseString = String.valueOf(log.responseCause());
+        } else {
+            responseCauseString = null;
+        }
+
+        final RequestContext ctx = log.context();
+        final String sanitizedHeaders;
+        if (availableProperties.contains(RequestLogProperty.RESPONSE_HEADERS)) {
+            sanitizedHeaders = responseHeadersSanitizer.apply(ctx, log.responseHeaders());
+        } else {
+            sanitizedHeaders = null;
+        }
+
+        final String sanitizedContent;
+        if (availableProperties.contains(RequestLogProperty.RESPONSE_CONTENT) &&
+            log.responseContent() != null) {
+            sanitizedContent = responseContentSanitizer.apply(ctx, log.responseContent());
+        } else if (availableProperties.contains(RequestLogProperty.RESPONSE_CONTENT_PREVIEW) &&
+                   log.responseContentPreview() != null) {
+            sanitizedContent = responseContentSanitizer.apply(ctx, log.responseContentPreview());
+        } else {
+            sanitizedContent = null;
+        }
+
+        final String sanitizedTrailers;
+        if (availableProperties.contains(RequestLogProperty.RESPONSE_TRAILERS) &&
+            !log.responseTrailers().isEmpty()) {
+            sanitizedTrailers = responseTrailersSanitizer.apply(ctx, log.responseTrailers());
+        } else {
+            sanitizedTrailers = null;
+        }
+
+        try (TemporaryThreadLocals tempThreadLocals = TemporaryThreadLocals.acquire()) {
+            final StringBuilder buf = tempThreadLocals.stringBuilder();
+            buf.append("{startTime=");
+            TextFormatter.appendEpochMicros(buf, log.responseStartTimeMicros());
+
+            if (availableProperties.contains(RequestLogProperty.RESPONSE_LENGTH)) {
+                buf.append(", length=");
+                TextFormatter.appendSize(buf, log.responseLength());
+            }
+
+            if (availableProperties.contains(RequestLogProperty.RESPONSE_END_TIME)) {
+                buf.append(", duration=");
+                TextFormatter.appendElapsed(buf, log.responseDurationNanos());
+                buf.append(", totalDuration=");
+                TextFormatter.appendElapsed(buf, log.totalDurationNanos());
+            }
+
+            if (responseCauseString != null) {
+                buf.append(", cause=").append(responseCauseString);
+            }
+
+            if (sanitizedHeaders != null) {
+                buf.append(", headers=").append(sanitizedHeaders);
+            }
+
+            if (sanitizedContent != null) {
+                buf.append(", content=").append(sanitizedContent);
+            }
+
+            if (sanitizedTrailers != null) {
+                buf.append(", trailers=").append(sanitizedTrailers);
+            }
+            buf.append('}');
+
+            final int numChildren = log.children() != null ? log.children().size() : 0;
```

Comment:
`log.children()` copies all elements whenever it is called. Should we declare a local variable?